internal/agent/device/applications/applications.go

```diff
@@ -0,0 +1,358 @@
+	case isRunningHealthy(total, healthy, initializing):
+		newStatus = v1alpha1.ApplicationStatusRunning
+		summary = v1alpha1.ApplicationsSummaryStatusHealthy
+	default:
+		panic(fmt.Sprintf("unexpected status: %d %d %d", total, healthy, initializing))
+	}
+
+	if a.status.Status != newStatus {
+		a.status.Status = newStatus
+	}
+	if a.status.Ready != readyStatus {
+		a.status.Ready = readyStatus
+	}
+	if a.status.Restarts != restarts {
+		a.status.Restarts = restarts
+	}
+
+	return a.status, summary
+}
+
+func isStarting(total, healthy, initializing int) bool {
+	return total > 0 && initializing > 0 && healthy > 0
+}
+
+func isUnknown(total, healthy, initializing int) bool {
+	return total == 0 && healthy == 0 && initializing == 0
+}
+
+func isPreparing(total, healthy, initializing int) bool {
+	return total > 0 && healthy == 0 && initializing > 0
+}
+
+func isRunningDegraded(total, healthy, initializing int) bool {
+	return total != healthy && healthy > 0 && initializing == 0
+}
+
+func isRunningHealthy(total, healthy, initializing int) bool {
+	return total > 0 && healthy == total && initializing == 0
+}
+
+func isErrored(total, healthy, initializing int) bool {
+	return total > 0 && healthy == 0 && initializing == 0
+}
+
+func ParseAppType(s string) (AppType, error) {
+	appType := AppType(s)
+	if !appType.isValid() {
+		return "", fmt.Errorf("invalid app type: %s", s)
+	}
+	return appType, nil
+}
+
+func (a AppType) isValid() bool {
+	switch a {
+	case AppCompose:
+		return true
+	default:
+		return false
+	}
+}
+
+func (a AppType) ActionHandler() (lifecycle.ActionHandlerType, error) {
+	switch a {
+	case AppCompose:
+		return lifecycle.ActionHandlerCompose, nil
+	default:
+		return "", fmt.Errorf("%w: %s", ErrorUnsupportedAppType, a)
+	}
+}
+
+type applications struct {
+	images []*application[*v1alpha1.ImageApplicationProvider]
+	// add other types of application providers here
+}
+
+func (a *applications) ImageBased() []*application[*v1alpha1.ImageApplicationProvider] {
+	return a.images
+}
+
+// ImageProvidersFromSpec returns a list of image application providers from a rendered device spec.
+func ImageProvidersFromSpec(spec *v1alpha1.RenderedDeviceSpec) ([]v1alpha1.ImageApplicationProvider, error) {
+	var providers []v1alpha1.ImageApplicationProvider
+	for _, appSpec := range *spec.Applications {
+		appProvider, err := appSpec.Type()
+		if err != nil {
+			return nil, err
+		}
+		if appProvider == v1alpha1.ImageApplicationProviderType {
+			provider, err := appSpec.AsImageApplicationProvider()
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert application to image provider: %w", err)
+			}
+			providers = append(providers, provider)
+		}
+	}
+	return providers, nil
+}
+
+func TypeFromImage(ctx context.Context, podman *client.Podman, image string) (AppType, error) {
+	labels, err := podman.InspectLabels(ctx, image)
+	if err != nil {
+		return "", err
+	}
+	appTypeLabel, ok := labels[AppTypeLabel]
```

Comment:
Do we have the format documented anywhere?

Reply:
no docs all users must read code :).  docs will be follow-up :pray: